Integrations/MimecastV2/MimecastV2.py

```diff
@@ -1479,6 +1479,101 @@ def download_attachment_request(attachment_id):
+    mail_from_ = demisto.args().get('mail_from', '').encode('utf-8')
+    subject = demisto.args().get('subject', '').encode('utf-8')
+    message_id = demisto.args().get('message_id', '').encode('utf-8')
+    limit = int(demisto.args().get('limit', 100))
+
+    if search_reason:
+        search_params['searchReason'] = search_reason
+    if start:
+        search_params['start'] = start
+    if end:
+        search_params['end'] = end
+    if message_id:
+        search_params['messageId'] = message_id
+    if mail_from_:
+        advanced_params['from'] = mail_from_
+    if mail_to:
+        advanced_params['to'] = mail_to
+    if subject:
+        advanced_params['subject'] = subject
+    if sender_ip:
+        advanced_params['senderIP'] = sender_ip
+
+    search_params['advancedTrackAndTraceOptions'] = advanced_params
+    tracked_mails = search_mail_request(search_params)
+    if limit:
+        tracked_mails = tracked_mails[:limit]
+
+    for tracked_mail in tracked_mails:
+        to = []  # type: List[Any]
+        receivers = tracked_mail.get('to', [])
+        for receiver in receivers:
+            to.append({
+                'DisplayableName': receiver.get('displayableName'),
+                'EmailAddress': receiver.get('emailAddress')
+            })
+
+        contents.append({
+            'Status': tracked_mail.get('status'),
+            'Received': tracked_mail.get('received'),
+            'FromEnv': {
+                'DisplayableName': tracked_mail.get('fromEnv', {}).get('displayableName'),
+                'EmailAddress': tracked_mail.get('fromEnv', {}).get('emailAddress')
+            },
+            'FromHdr': {
+                'DisplayableName': tracked_mail.get('FromHdr', {}).get('displayableName'),
+                'EmailAddress': tracked_mail.get('FromHdr', {}).get('emailAddress')
+            },
+            'Attachments': tracked_mail.get('attachments'),
+            'To': to,
+            'SenderIP': tracked_mail.get('senderIP'),
+            'Route': tracked_mail.get('route'),
+            'ID': tracked_mail.get('id'),
+            'Sent': tracked_mail.get('sent'),
+            'Subject': tracked_mail.get('subject')
+        })
+
+    context['Mimecast.TrackedEmails'] = contents
+
+    results = {
+        'Type': entryTypes['note'],
+        'ContentsFormat': formats['json'],
+        'Contents': contents,
+        'ReadableContentsFormat': formats['markdown'],
+        'HumanReadable': tableToMarkdown('Mimecast Search: ', contents, headers),
+        'EntryContext': context
+    }
+
+    return results
+
+
+def search_mail_request(search_params):
+    # Setup required variables
+    api_endpoint = '/api/message-finder/search'
+    payload = {
+        'data': [search_params]
+    }
+
+    response = http_request('POST', api_endpoint, str(payload))
+    if response.get('fail'):
+        return_error(json.dumps(response.get('fail')[0].get('errors')))
```

Comment:
lets verify that the output of `json.dumps(response.get('fail')` is a list is not empty and that the output is a dict